app/(frontend)/(app)/profile/page.tsx

```diff
@@ -1,7 +1,116 @@
+"use client"
+
+import {Avatar} from "@nextui-org/avatar";
+import {useCallback, useEffect, useState} from "react";
+import useQuery, {IQueryParams} from "@/app/ui/hooks/useQuery";
+import {ITweet} from "@/app/lib/definitions";
+import Tweet from "@/app/ui/Tweet";
+
+type ProfileData = {
+    profilePicture: string
+    name: string
+    displayName: string
+    id: string
+    description: string
+    createdAt: Date
+}
+
 export default function ProfilePage() {
+
+    const query$ = useQuery()
+
+    const [profileData, setProfileData] = useState<ProfileData>()
+    const [tweets, setTweets] = useState<ITweet[]>([])
+
+    useEffect(() => {
+        const userId = localStorage.getItem('currentUserId')
+        const profileParams: IQueryParams = {
+            URL: `/profile/${userId}`,
+            method: 'GET',
+            authorized: true
+        }
+
+        const tweetsParams: IQueryParams = {
+            URL: `/tweets?userId=${userId}`,
+            method: 'GET',
+            authorized: true
+        }
+
+        query$(profileParams).subscribe({
+            next: ({data: {profile}}) => {
+                setProfileData(profile)
+            },
+            error: err => {
+                console.error(err)
+            }
+        })
+
+        query$(tweetsParams).subscribe({
```

Comment:
It would be better to combine these two observables:
- `switchMap` is a `pipe`
- it allows you to process the data from an observable just like in a `subscription`
- after that, you can return the `next observable` _(the previous subscription will be "cancelled", so you won't receive data from that, but in this case, it's not relevant, just a fun fact)_
- the benefit is that you manage two subscriptions in one, and it is really useful when one subscription relies on the other
- for example we might not want to load the tweets when there's no profile for that user
[Documentation](https://www.learnrxjs.io/learn-rxjs/operators/transformation/switchmap)